jena-shacl/src/main/java/org/apache/jena/shacl/engine/constraint/PatternConstraint.java

```diff
@@ -53,6 +53,18 @@ public PatternConstraint(String pattern, String flagsStr) {
         this.pattern = Pattern.compile(pattern, flags);
     }
 
+    public Pattern getPattern() {
```

Comment:
Minor: I'd prefer not the Pattern because it is an internal term, and then have the pattern string as `getPattern()`.

It is exposing the information once, not in two different ways.

Reply:
ok